launchpad-stack-aws3/infra/variables.tf

```diff
@@ -31,13 +31,14 @@ variable "network" {
 variable "nodegroups" {
   description = "A map of machine group definitions"
   type = map(object({
-    platform    = string
-    type        = string
-    count       = number
-    volume_size = number
-    role        = string
-    public      = bool
-    user_data   = string
+    platform      = string
+    type          = string
+    count         = number
+    volume_size   = number
+    role          = string
+    public        = bool
+    user_data     = string
+    platform_type = string
```

Comment:
I was also having a hard time figuring out how to identify platform_type, but I am not convinced that making the user do the work, instead of pulling the information from the "platform" module, either from the static list of platforms, or from the ami data.

Reply:
Changes made to get platform_type from existing data pulled from platform module.